lpcompiler.pas

```diff
@@ -1310,7 +1325,17 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
         end;
       until (Tokenizer.Tok = tk_sym_ParenthesisClose);
 
-    if isFunction then
+    if Result.isOperator then
+    begin
+      if (Result.Params.Count <> 2) then
+        LapeExceptionFmt(lpeInvalidOperator, [op_name[op], 2], Pos);
+      ltyp := Result.Params[0].VarType;
+      rtyp := Result.Params[1].VarType;
+      if ValidEvalFunction(GetEvalProc(op, ltyp.BaseType, rtyp.BaseType)) then
```

Comment:
It might be better to use ltype.resType() here as GetEvalProc only works for base types.